### ISSUE TITLE
Reversed pixel coordinates when initializing with Sky Coordinate

### ISSUE THREAD

**Issue by charlotteaward** [pmelchior/scarlet2#91]
When I have some Sky Coordinates built from the (verified) RAs and Decs of sources in an image, and initialize Sources using those Sky Coordinates, the sources are initialized in the wrong place, with inverted x and y pixel coordinates (see image at end). Digging into it, I think this is because `pos.to_pixel(wcs_)` of line 56 of frame.py, in the `get_pixel` function returns the x and y coordinates that match the observation data, but in parts of the scarlet2 code it seems to be assuming that those pixel coordinates should be reversed, e.g. in source.py line 43. Its not a model frame issue, because if you do `pos.to_pixel(wcs_)` where `wcs_` is from the model frame or the observation frame, the pixel coordinates match the source locations in the image. But the sources are clearly rendered in the wrong place, and the boxes are shown to be in the wrong place when plotting. 

I had (quite unfortunately!) been avoiding this issue by introducing another 'bug' that cancelled out this problem: I was finding the RAs and Decs that matched the inverted pixel coordinates by also reversing the pixel coordinates of the sep source detection output, and using those for source initialization. But my collaborator noticed that those intermediate RAs and Decs were incorrect. 

So, in summary, source initialization using Sky Coordinates seems to treating the pixel coordinates inconsistently, sometimes inverting them and sometimes not. (Unless I have some even weirder issues with ZTF wcs's or there's a confusing bug in the plotting code).

![image](https://github.com/user-attachments/assets/828d95c9-2dbe-4018-a611-b340e798cce8)

**Comment by charlotteaward** [comment]
The fix seems to be to change line 56 in frame.py to make sure it reverses the pixel coordinates. The transpose operation doesn't do this so we can return pixel[::-1] instead of pixel or something like that. Will make a PR for testing.

**Comment by b-remy** [comment]
Closing as fixed in #94 !
